tests/hooks/test_dbapi.py

```diff
@@ -232,3 +233,74 @@ def handler(cur):
         assert called == 2
         assert self.conn.commit.called
         assert result == [obj, obj]
+
+    @patch('airflow.hooks.dbapi.DbApiHook._run_command')
+    def test_run_with_multiple_statements_and_split(self, _run_command):
+        _cases = [
+            (['SQL1; SQL2;', 'SQL3'], 3),
+            (['SQL1; SQL2;', 'SQL3;'], 3),
+            (['SQL1; SQL2; SQL3;'], 3),
+            ('SQL1; SQL2; SQL3;', 3),
+            (['SQL1;', 'SQL2'], 2),
+            (['SQL1;', 'SQL2;'], 2),
+            (['SQL1; SQL2;'], 2),
+            ('SQL1; SQL2;', 2),
+            (['SQL1'], 1),
+            (['SQL1;'], 1),
+            ('SQL1;', 1),
+            ('SQL1', 1),
+            (
+                """
+            CREATE OR REPLACE FUNCTION dfp.extract_vcode(NAME VARCHAR)
+                RETURNS string
+                LANGUAGE javascript
+                STRICT
+                AS '
+                const regex = /[[]v=([0-9]+)/ig;
+                let s = NAME.match(regex);
+                if (s != null) {
+                    return s[0].split('=')[1];
+                } else {
+                    return null;
+                }
+                ';
+                CREATE OR REPLACE FUNCTION dfp.parse_metadata(DATA varchar)
+                RETURNS OBJECT
+                LANGUAGE javascript
+                STRICT
+                AS '
+                if (!DATA) {
+                    return {}
+                }
+
+                let dict = {}
+                const parts = DATA.split("|")
+
+                parts.forEach((p) => {
+                    const split = p.split("=")
+                    dict[split[0]] = split[1]
+                })
+
+                return dict
+                ';
+            """,  # nopep8
+                2,
+            ),
+            (
+                """
+            SELECT
+                *
+            FROM country
+            LEFT JOIN city ON city.country_id = country.id
+            LEFT JOIN customer ON city.id = customer.city_id
+            LEFT JOIN call ON call.customer_id = customer.id;
+            SELECT country, count(*) FROM country
+            GROUP BY 1;
+            """,  # nopep8
```

Comment:
Why is this `nopep8` comment needed?

Reply:
Also this entire `for case in cases` thing should be done with `parameterized` instead. And I feel we should check the actual parsed SQL, not just the count.

Reply:
1) because the static checks fails without that because of E501 (long multiline string), if there is a better way, im open to change it.

2) I think that is a good improvement for the test. Will do at the end of the day.

Reply:
This sounds wrong. Airflow sets the line length to 110 and this is nowhere near that.

https://github.com/apache/airflow/blob/main/.flake8

Is this reported by pre-commit? Or maybe your editor is not picking up the configuration correctly?

Reply:
I think in many editors you have to set it manually (Intellij https://intellij-support.jetbrains.com/hc/en-us/community/posts/115000617684-Configuring-PEP8-not-ignoring-rules-really-changing-it-#:~:text=The%20PEP8%20limit%20in%20PyCharm,(and%2072%20for%20comments).) 

PEP8 is flexible around line length and indeed we "strongly prefer a bit longer line"